publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java

```diff
@@ -323,35 +327,84 @@ public ResponseEntity generateAndPublish(final String msgProtocol, final String
+     * @param responseMessage the message according to response
+     * @param resultMessage whatever the result this message gives you idea about that
+     * @param errorResponse is to collect all the responses here.
+     * @return ResponseEntity
+     */
+    public ResponseEntity<JsonObject> createResponseEntity(HttpStatus status, String responseMessage, String resultMessage,
+                                                           JsonObject errorResponse) {
+        initializeResponse(status, responseMessage, resultMessage, errorResponse);
+        return new ResponseEntity<>(errorResponse, status);
+    }
+
+    /**
+     * To display response in browser or application
+     * @param status response code for the HTTP request
+     * @param responseMessage the message according to response
+     * @param resultMessage whatever the result this message gives you idea about that
+     * @return ResponseEntity
+     */
+    public ResponseEntity<JsonObject> createResponseEntity(HttpStatus status, String responseMessage, String resultMessage) {
+        return createResponseEntity(status, responseMessage, resultMessage, new JsonObject());
+    }
+
+    /**
+     * To initialize in the @{createResponseEntity} method
+     * @param status response code for the HTTP request
+     * @param resultMessage whatever the result this message gives you idea about that
+     * @param errorResponse is to collect all the responses here.
+     */
+    public void initializeResponse(HttpStatus status, String errorMessage, String resultMessage, JsonObject errorResponse) {
+        errorResponse.addProperty(JSON_STATUS_CODE, status.value());
+        errorResponse.addProperty(JSON_STATUS_RESULT, resultMessage);
+        errorResponse.addProperty(JSON_EVENT_MESSAGE_FIELD, errorMessage);
+    }
+
+    /**
+     * This one is helper method to process or publish messages or event
+     * @param eventResponseMessage
      * @param msgProtocol
      * @param userDomain
      * @param tag
      * @param routingKey
      * @return list of responses
      */
-    public List<Map<String, Object>> processingValidEvent(String eventMsg, final String msgProtocol, final String userDomain,
+    public List<Map<String, Object>> processingValidEvent(String eventResponseMessage, final String msgProtocol, final String userDomain,
                                                           final String tag, final String routingKey) {
         MsgService msgService = PublishUtils.getMessageService(msgProtocol, msgServices);
         List<Map<String, Object>> responseEvent = new ArrayList<>();
-        Map<String, Object> eventResponse;
-        JsonElement eventElement = JsonParser.parseString(eventMsg);
-        for (int i = 0; i < eventElement.getAsJsonArray().size(); i++) {
+        Map<String, Object> eventResponse = null;
+
+        if (eventResponseMessage == null) {
+            eventResponse.put("Parameter 'eventResponseMessage' must not be null", HttpStatus.BAD_REQUEST);
```

Comment:
This is response to end user. How the user knows what 'eventResponseMessage' is?

Reply:
I'd write something like 'Missing response from 'generate' service.'